rust/headless-client/src/ipc_service.rs

```diff
@@ -371,7 +376,10 @@ impl<'a> Handler<'a> {
                     Session::connect(args, portal, tokio::runtime::Handle::try_current()?);
                 let tun = self.tun_device.make_tun()?;
                 new_session.set_tun(Box::new(tun));
-                new_session.set_dns(dns_control::system_resolvers().unwrap_or_default());
+                new_session.set_dns(
+                    dns_control::system_resolvers(self.dns_controller.dns_control_method)
+                        .unwrap_or_default(),
```

Comment:
This looks a bit convoluted. Can we do something like this instead?

```suggestion
                    self.dns_controller.system_resolvers(),
```